linker/descriptors.go

```diff
@@ -1153,9 +1152,21 @@ func (f *fldDescriptor) TextName() string {
 	if f.IsExtension() {
 		return fmt.Sprintf("[%s]", f.FullName())
 	}
+	if f.looksLikeGroup() {
+		// groups use the type name
+		return string(protoreflect.FullName(f.proto.GetTypeName()).Name())
+	}
 	return string(f.Name())
 }
 
+func (f *fldDescriptor) looksLikeGroup() bool {
+	// It looks like a group if it uses group/delimited encoding (checked via f.Kind)
+	// and the message type is a sibling whose name is a mixed-case version of the field name.
+	return f.Kind() == protoreflect.GroupKind &&
+		f.Message().FullName().Parent() == f.FullName().Parent() &&
+		string(f.Name()) == strings.ToLower(string(f.Message().Name()))
```

Comment:
Could `f.Name()` not be mixed case for a group? Assuming it's fine.

Reply:
It could be all upper-case. But the compiler _requires_ the group name to start with a capital letter, so it is almost always mixed-case due use of `PascalCase` naming convention. The corresponding field name is never identical to the type name, because the type contains at least one capital letter whereas the field name is always lower-case.